pkg/ldd/ldd_unix.go

```diff
@@ -70,22 +70,34 @@ func follow(l string, names map[string]*FileInfo) error {
 	}
 }
 
-// runinterp runs the interpreter with the --list switch
-// and the file as an argument. For each returned line
-// it looks for => as the second field, indicating a
-// real .so (as opposed to the .vdso or a string like
-// 'not a dynamic executable'.
+func LdSo() (string, error) {
+	n, err := filepath.Glob(ldso)
+	if err != nil {
+		return "", err
+	}
+	if len(n) == 0 {
+		return "", fmt.Errorf("No ld.so matches %v", ldso)
+	}
+	return n[0], nil
+}
+
+// runinterp runs the interpreter on the given file
+// and returns a list of shared libs it depends on.
 func runinterp(interp, file string) ([]string, error) {
 	var names []string
-	o, err := exec.Command(interp, "--list", file).Output()
-	if err != nil {
+	o, err := lddOutput(interp, file)
+	if err != nil && o == nil {
```

Comment:
This test is not needed, if o is nil the loop below will exit, see
https://play.golang.org/p/naYxZlmzIFK

Reply:
I mean the additional test, test for err != nil is fine.